### PR TITLE
Require application/octet-stream content type for txs [ECR-2650]

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
## Overview

Require application/octet-stream content type for submitting
transactions. application/x-www-form-urlencoded does not work
for unescaped binary data.

---
See: 
 - The discussion in https://jira.bf.local/browse/ECR-2650
 - Reproducer: #580 
 - FE fix: #621 

### Definition of Done

- [X] There are no TODOs left in the code
- [X] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [X] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [X] Public API has Javadoc
- [X] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes
- [x] **frontend is updated to set the header properly**

### CODE REVIEW

File: exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/ApiController.java [hunk 1]
```diff
@@ -83,6 +83,9 @@ void mountApi(Router router) {
   }
 
   private void submitTransaction(RoutingContext rc) {
+    checkRequestContentType(rc, "application/octet-stream");
```
Comment (DENMROOT) [inline exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/ApiController.java]: Is it required for all Exonum Services implementation ? Should be specified.

Comment (dmitry-timofeev) [reply exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/ApiController.java]: Not sure I understand what you mean. This code is in a particular service implementation, in a particular endpoint. It is not a part of the framework.

File: exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/ApiController.java [hunk 2]
```diff
@@ -97,6 +100,14 @@ private void submitTransaction(RoutingContext rc) {
         .end(String.valueOf(txHash));
   }
 
+  private void checkRequestContentType(RoutingContext rc,
+      String expectedContentType) {
+    String contentType = rc.request().getHeader("Content-Type");
+
+    checkArgument(expectedContentType.equals(contentType), "Content-Type must be %s, but %s",
```
Comment (bullet-tooth) [inline exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/ApiController.java]:
```suggestion
    checkArgument(expectedContentType.equals(contentType), "Content-Type must be %s, but was %s",
```

File: exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/ApiController.java (hunk 1 above)
Comment (vitvakatu) [reply exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/ApiController.java]: I think @DENMROOT meant that we might need to fix cryptocurrency frontend as well.

Comment (dmitry-timofeev) [reply exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/ApiController.java]: We won't integrate that till front is updated: https://jira.bf.local/browse/ECR-2650?focusedCommentId=30419&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-30419

File: exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/ApiController.java [hunk 3]
```diff
@@ -97,6 +100,14 @@ private void submitTransaction(RoutingContext rc) {
         .end(String.valueOf(txHash));
   }
 
+  private void checkRequestContentType(RoutingContext rc,
+      String expectedContentType) {
```
Comment (MakarovS) [inline exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/ApiController.java]: Do we need a new line here?

**Pull request merged**